### PR TITLE
Removing assignInstance from V2 route

### PR DESCRIPTION
**Pull request by subhankarc**
As the factory pattern is no more used for lifecycle operations,

* Removed the assignInstance middleware from V2
* changed the corresponding code in the EventLogIntercepter and FabrikBaseController
* Added a test for Director Dashboard
* Removed unneeded code from `DirectorManager` and `DirectorInstance`.

### CODE REVIEW

File: api-controllers/routes/broker/v2.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ router.use(commonMiddleware.error({
 
 /* Service Instance Router */
 instanceRouter.use(controller.handler('ensurePlatformContext'));
-instanceRouter.use(controller.handler('assignInstance'));
+//instanceRouter.use(controller.handler('assignInstance'));
```
Comment (kamath-prasad) [inline api-controllers/routes/broker/v2.js]: remove commented code.

File: broker/lib/fabrik/DirectorInstance.js [hunk 2]
```diff
@@ -23,28 +23,6 @@ class DirectorInstance extends BaseInstance {
     this.networkSegmentIndex = undefined;
```
Comment (kamath-prasad) [inline broker/lib/fabrik/DirectorInstance.js]: similar changes not required for docker - https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/broker/lib/fabrik/DockerInstance.js#L36-L58 ?

Comment (subhankarc) [reply broker/lib/fabrik/DirectorInstance.js]: Oops! I made a mistake. Actually this cannot be removed. DashboardController uses it still!

**Comment by subhankarc** [comment]
@kamath-prasad The factory pattern in the V2 route is removed with this. But it cannot be immediately removed as Dashboard and extension routes still uses them.

**Comment by subhankarc** [comment]
@kamath-prasad There were no test for director dashboard access. Added a test which catches accidental removal of code like this.

**Comment by kamath-prasad** [comment]
@subhankarc : ok , can you please update the tests ? coverage reduced ..

**Pull request merged**